test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java

```diff
@@ -293,18 +293,21 @@ void checkWarningHeaders(final List<String> warningHeaders, final Version master
             if (matches) {
                 final String message = matcher.group(1);
                 // noinspection StatementWithEmptyBody
-                if (masterVersion.before(Version.V_7_0_0)
+                if ((masterVersion.before(Version.V_7_0_0)
                         && message.equals("the default number of shards will change from [5] to [1] in 7.0.0; "
                         + "if you wish to continue using the default of [5] shards, "
-                        + "you must manage this on the create index request or with an index template")) {
+                        + "you must manage this on the create index request or with an index template"))
+                    || message.startsWith("[types removal]")) {
```

Comment:
Oops, I indeed misread the parentheses ; I double checked even but it’s exactly because I had to strain across multiple lines to match them up I propose the change. :innocent: